### PR TITLE
Quick start ND4J

### PR DESCRIPTION
**Pull request by RobAltena**
Replace he mess made in #7911 .

### CODE REVIEW

File: docs/nd4j/templates/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+* A versatile n-dimensional array object.
+* Linear algebra and signal processing functions.
+* Multiplatform functionality including GPUs.
+
+This quickstart follows the same layout and approach of the [Numpy quickstart](https://docs.scipy.org/doc/numpy/user/quickstart.html). This should help people familiar with Python and Numpy get started quickly with Nd4J.
+
+## Prerequisites
+<!--- // Java, Maven, git. Coding skills and hello world example. -->
+
+To follow the examples in this quick start you will need to know some Java. You will also need to install the following software on your computer:
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: "need to know some Java".
Let's also briefly mention that ND4J can be used in any JVM language (scala, kotlin, etc)

File: docs/nd4j/templates/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+<!--- Comments are standard html. Tripple dash based on stackoverflow: https://stackoverflow.com/questions/4823468/comments-in-markdown -->
+
+<!--- Borrowing the layout of the Numpy quickstart to get started. -->
+
+## Introduction
+<!--- What is ND4J and why is it important. From the nd4j repo readme.  -->
+ND4J is a scientific computing library for the JVM. It is meant to be used in production environments rather than as a research tool, which means routines are designed to run fast with minimum RAM requirements. The main features are:
+* A versatile n-dimensional array object.
+* Linear algebra and signal processing functions.
+* Multiplatform functionality including GPUs.
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Multi-platform - or maybe a separate point - let's mention all major OSs (win/linux/osx/android), and chips (x86, arm).

File: docs/nd4j/templates/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+<!--- git allows us to start with a cleaner project than mvn create. -->
+* [Git](./deeplearning4j-quickstart#Git)(distributed version control system)
+
+If you are confident you know how to use maven and git, please feel free to skip to the [Basics](#Basics). In the remainder of this section we will build a small 'hello ND4J' application to verify the prequisites are set up correctly.
+
+Execute the following commands to get the project from github. 
+
+<!--- TODO: Create HelloNd4J or Quickstart-nd4j repo in Deeplearning4J. -->
+```shell
+git clone https://github.com/RobAltena/HelloNd4J.git
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Good idea. Eventually let's put this under deeplearning4j repo.

File: docs/nd4j/templates/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+mvn exec:java -Dexec.mainClass="HelloNd4j"
+```
+
+When everything is set up correctly you should see the following output:
+
+```shell
+SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
+SLF4J: Defaulting to no-operation (NOP) logger implementation
+SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
+[         0,         0]
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Add logger to repo.

File: docs/nd4j/templates/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+While this quickstart is being build, please refer to our existing 
+[basics usage](./nd4j-basics) document.
+
+The main feature of Nd4j is the versatile n-dimensional array interface called INDArray. To improve performance Nd4j uses off-heap memory to store data. The INDArray is different from standard Java arrays.
+
+Some of the key properties and methods for an INDArray x are as follows:
+
+```java
+// The number of axes (dimensions) of the array.
+int dimensions = x.shape().length;
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: x.rank() instead of x.shape().length

File: docs/nd4j/templates/quickstart.md [hunk 6]
```diff
@@ -0,0 +1,381 @@
...
+
+// The dimensions of the array. The size in each dimension.
+long[] shape = x.shape();
+
+// The total number of elements.
+length = x.length();
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: `long length = ...`

File: docs/nd4j/templates/quickstart.md [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+### Array Creation
+To create INDArrays you use the static factory methods of the [Nd4j](https://deeplearning4j.org/api/latest/org/nd4j/linalg/factory/Nd4j.html ) class.
+
+<!--- We have good docs on creating INDArrays already.  -->
+<!--- https://deeplearning4j.org/docs/latest/nd4j-overview#creating -->
+The `Nd4j.create` function is overloaded to make it easy to create INDArrays from regular Java arrays. The example below uses Java `double` arrays. Similar create methods are overloaded for `float`, `int` and `long`.
+
+```java
+double arr_2d[][]={{1.0,2.0,3.0},{4.0,5.0,6.0},{7.0,8.0,9.0}};
+INDArray x_2d = Nd4j.create(arr_2d);
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Let's direct users to Nd4j.createFromArray instead.
Unlike Nd4j.create, there's overloads for all java primitive types and Number (long, short, Long, Boolean etc), and so users are less likely to get confused with Nd4j.create(int[]), where they think the int[] is content just like Nd4j.create(double[]), but it's actually shape

File: docs/nd4j/templates/quickstart.md [hunk 8]
```diff
@@ -0,0 +1,381 @@
...
+double arr_1d[]={1.0,2.0,3.0};
+INDArray  x_1d = Nd4j.create(arr_1d);
+```
+
+There are overloaded `create` functions for INDArrays up to 4 dimensions. To create INDArrays with any shape use the `create` functions that take a shape as one of their arguments.
+
+```java
+double[] flat = ArrayUtil.flattenDoubleArray(myDoubleArray);
+int[] shape = ...;	//Array shape here
+INDArray myArr = Nd4j.create(flat,shape,'c');
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: This is old. Nd4j.createFromArray has overloads up to 4d for all types

File: docs/nd4j/templates/quickstart.md [hunk 9]
```diff
@@ -0,0 +1,381 @@
...
+
+Nd4j can create arrays initialized with zeros and ones using the functions `zeros` and `ones`. The `rand` function allows you to create an array initialized with random values.
+The default datatype of the INDarray created is `float`. Some overloads allow you to set the datatype.
+
+```java
+INDArray  x = Nd4j.zeros(5);
+//[         0,         0,         0,         0,         0], FLOAT
+
+int [] shape = {5};
+x = Nd4j.zeros(shape, DataType.DOUBLE);
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Use `Nd4j.zeros(DataType.DOUBLE, 5)`,  no need to do it in 2 lines with array for shape.
Maybe also show the import for DataType the first time it's used.

File: docs/nd4j/templates/quickstart.md [hunk 10]
```diff
@@ -0,0 +1,381 @@
...
+//[         0,         0,         0,         0,         0], DOUBLE
+
+// For higher dimensions you can provide a shape array. 2D random matrix example:
+int rows = 4;
+int cols = 5;
+int[] shape = {rows, cols};
+INDArray x = Nd4j.rand(shape);
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: hm... keey it for now, but eventually we'll switch to varargs once this is done: https://github.com/eclipse/deeplearning4j/issues/7950

File: docs/nd4j/templates/quickstart.md [hunk 11]
```diff
@@ -0,0 +1,381 @@
...
+// [         0,    1.0000,    2.0000,    3.0000,    4.0000]
+
+INDArray  x = Nd4j.arange(2, 7);
+// [    2.0000,    3.0000,    4.0000,    5.0000,    6.0000]
+```
+
+The `linspace` function allows you to specify the number of points generated:
+```java
+INDArray  x = Nd4j.linspace(1, 10, 5);
+// [    1.0000,    3.2500,    5.5000,    7.7500,   10.0000]
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Maybe add comment for args order: (start, stop, count)

File: docs/nd4j/templates/quickstart.md [hunk 12]
```diff
@@ -0,0 +1,381 @@
...
+[[[         0,    1.0000,    2.0000,    3.0000], 
+  [    4.0000,    5.0000,    6.0000,    7.0000], 
+  [    8.0000,    9.0000,   10.0000,   11.0000]], 
+
+ [[   12.0000,   13.0000,   14.0000,   15.0000], 
+  [   16.0000,   17.0000,   18.0000,   19.0000], 
+  [   20.0000,   21.0000,   22.0000,   23.0000]]]
+*/
+```
+
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Let's add a brief note - limited precision, and limited number of elements shown for large arrays.

File: docs/nd4j/templates/quickstart.md [hunk 13]
```diff
@@ -0,0 +1,381 @@
...
+### Basic Operations
+You will have to use INDArray methods to perform operations on your arrays. There are  in-place and copy overloads and scalar and element wise overloaded versions. The in-place operators return a reference to the array so you can conveniently chain operations together.
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Maybe add one sentence on performance implications: basically, "use inplace operations where possible, non-inplace ops have new array creation overhead"

File: docs/nd4j/templates/quickstart.md [hunk 14]
```diff
@@ -0,0 +1,381 @@
...
+division: arr.div(...), arr.divi(...)
+
+When you perform the basic operations you must make sure the underlying data types are the same.
+```java
+int [] shape = {5};
+INDArray  x = Nd4j.zeros(shape, DataType.DOUBLE);
+INDArray  x2 = Nd4j.zeros(shape, DataType.INT);
+INDArray  x3 = x.add(x2);
+// java.lang.IllegalArgumentException: Op.X and Op.Y must have the same data type, but got INT vs DOUBLE
+```
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Show casting example as solution to this:
`INDArray x3 = x.add(x2.cast(DataType.DOUBLE))`

File: docs/nd4j/templates/quickstart.md [hunk 15]
```diff
@@ -0,0 +1,381 @@
...
+<!--- Moving matrix operations after Transforms. In Nd4j the dot product is a transform. -->
+
+The IndArray has methods implementing operations such as `sum`, `min`, `max`.
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: IndArray -> INDArray
"operations such as" - mention they are called reduction/accumulation operations.

File: docs/nd4j/templates/quickstart.md [hunk 16]
```diff
@@ -0,0 +1,381 @@
...
+0.1504
+0.9584
+*/
+```
+
+Provide a dimension argument to apply the operation across the specified dimension:
+
+```java
+int [] shape = {3,4};
+INDArray x = Nd4j.arange(12).reshape(shape);
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Reshape method is varargs, `Nd4j.arange(12).reshape(3,4);` Is cleaner IMO

File: docs/nd4j/templates/quickstart.md [hunk 17]
```diff
@@ -0,0 +1,381 @@
...
+
+<!--- The numpy quickstart calls them Universal Functions. -->
+### Transform operation
+Nd4j provides familiar mathematical functions such as sin, cos, and exp. These are called transform operations. The result is returned as an INDArray.
+
+```java
+INDArray x = Nd4j.arange(3);
+System.out.println(x);
+// [         0,    1.0000,    2.0000]
+System.out.println(exp(x));
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: I'd rather we don't rely on static imports here. It's nice and short, but user will copy/paste this, miss the static import, and then be confused.
```
import org.nd4j.linalg.ops.transforms.Transforms
System.out.println(Transforms.exp(x));
```

File: docs/nd4j/templates/quickstart.md [hunk 18]
```diff
@@ -0,0 +1,381 @@
...
+
+
+### Matrix multiplication
+We have already seen the element wise multiplcation in the basic operations. The other Matrix operations have their own methods:
+
+```java
+int[] shape = {3, 4};
+INDArray x = Nd4j.arange(12).reshape(shape);
+int[] shape2 = {4, 3};
+INDArray y = Nd4j.arange(12).reshape(shape2);
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: More reshape varargs

File: docs/nd4j/templates/quickstart.md [hunk 19]
```diff
@@ -0,0 +1,381 @@
...
+// dot product.
+INDArray x = Nd4j.arange(12);
+INDArray y = Nd4j.arange(12);
+System.out.println(dot(x, y));  
+//506.0000
+```
+
+### Indexing, Slicing and Iterating
+Indexing, Slicing and Iterating is harder in Java than in Python. 
+To retreive individual values from an INDArray you can use the `getDouble`, `getFloat` or `getInt` methods. INDArrays cannot be indexed like Java arrays. You can get a Java array from an INDArray using `.data().asDouble()`.
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: `.data().asDouble()` etc isn't something I want in the quickstart, it's confusing for new users due to views and f/c order (not to mention permuted arrays).
Instead, direct users to toFloatVector(), toDoubleMatrix() etc

File: docs/nd4j/templates/quickstart.md (hunk 10 above)
Comment (RobAltena) [reply docs/nd4j/templates/quickstart.md]: Just keeping this one. Fixing all others.

File: docs/nd4j/templates/quickstart.md (hunk 14 above)
Comment (RobAltena) [reply docs/nd4j/templates/quickstart.md]: `INDArray x3 = x.add(x2.castTo(DataType.DOUBLE));`

File: docs/nd4j/templates/quickstart.md (hunk 17 above)
Comment (RobAltena) [reply docs/nd4j/templates/quickstart.md]: I will stick the imports on top.

File: docs/nd4j/templates/quickstart.md [hunk 20]
```diff
@@ -0,0 +1,565 @@
...
+ [readTxt](https://deeplearning4j.org/api/latest/org/nd4j/linalg/factory/Nd4j.html#readTxt-java.lang.String- ),
+ [zeros](https://deeplearning4j.org/api/latest/org/nd4j/linalg/factory/Nd4j.html#zeros-int...- ), 
+ [zeros_like](https://deeplearning4j.org/api/latest/org/nd4j/linalg/factory/Nd4j.html#zerosLike-org.nd4j.linalg.api.ndarray.INDArray- )
+
+### Conversions
+[convertToDoubles](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ndarray/INDArray.html#convertToDoubles-- ), 
+[convertToFloats](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ndarray/INDArray.html#convertToFloats-- ), 
+[convertToHalfs](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ndarray/INDArray.html#convertToHalfs-- )
+
+### Manipulations
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: permute op/method

File: docs/nd4j/templates/quickstart.md [hunk 21]
```diff
@@ -0,0 +1,565 @@
...
+[transpose](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ndarray/INDArray.html#transpose-- ),
+[vstack](https://deeplearning4j.org/api/latest/org/nd4j/linalg/factory/Nd4j.html#vstack-org.nd4j.linalg.api.ndarray.INDArray...- )
+
+### Ordering
+[argmax](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ndarray/INDArray.html#argMax-int...- ),
+[max](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ndarray/INDArray.html#max-int...- ),
+[min](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ndarray/INDArray.html#min-int...- ),
+[sort](https://deeplearning4j.org/api/latest/org/nd4j/linalg/factory/Nd4j.html#sort-org.nd4j.linalg.api.ndarray.INDArray-int-boolean- )
+
+### Operations
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Maybe link to Transforms class, there's tons of op helpers there (and even that isn't exhaustive...)

File: docs/nd4j/templates/quickstart.md [hunk 22]
```diff
@@ -0,0 +1,565 @@
...
+<!--- Borrowing the layout of the Numpy quickstart to get started. -->
+
+## Introduction
+<!--- What is ND4J and why is it important. From the nd4j repo readme.  -->
+ND4J is a scientific computing library for the JVM. It is meant to be used in production environments rather than as a research tool, which means routines are designed to run fast with minimum RAM requirements. The main features are:
+* A versatile n-dimensional array object.
+* Linear algebra and signal processing functions.
+* Multiplatform functionality including GPUs.
+    * all major operating systems: win/linux/osx/android.
+    * architectures: x86, arm, ppc.
```
Comment (AlexDBlack) [inline docs/nd4j/templates/quickstart.md]: Let's add bullet point reference to SameDiff (but note it's not covered at all in this page...)

File: docs/nd4j/templates/quickstart.md [hunk 23]
```diff
@@ -0,0 +1,565 @@
...
+
+This quickstart follows the same layout and approach of the [Numpy quickstart](https://docs.scipy.org/doc/numpy/user/quickstart.html). This should help people familiar with Python and Numpy get started quickly with Nd4J.
+
+## Prerequisites
+<!--- // Java, Maven, git. Coding skills and hello world example. -->
+
+To follow the examples in this quick start you will need to know some Java. You can use Nd4J from any [JVM Language](https://en.wikipedia.org/wiki/List_of_JVM_languages). (For example: Scala, Kotlin). You will also need to install the following software on your computer:
+<!--- from the dl4j quickstart, pointing to the dl4j quiclstart for details. -->
+* [Java (developer version)](./deeplearning4j-quickstart#Java) 1.7 or later (Only 64-Bit versions supported)
+* [Apache Maven](./deeplearning4j-quickstart#Maven) (automated build and dependency manager)
```
[the remark this answers is not in the document]
Comment (RobAltena) [reply docs/nd4j/templates/quickstart.md]: This one is a bit challenging. Re-written the section. Keeping the conversation open,

File: docs/nd4j/templates/quickstart.md [hunk 24]
```diff
@@ -0,0 +1,565 @@
...
+```shell
+[         0,         0]
+```
+
+## Basics
+<!--- TODO: We will put some into this page. Start with refering to existing doc. -->
+While this quickstart is being build, please refer to our existing 
+[basics usage](./nd4j-basics) document.
+
+The main feature of Nd4j is the versatile n-dimensional array interface called INDArray. To improve performance Nd4j uses off-heap memory to store data. The INDArray is different from standard Java arrays.
```
[the remark this answers is not in the document]
Comment (RobAltena) [reply docs/nd4j/templates/quickstart.md]: https://deeplearning4j.org/docs/latest/deeplearning4j-config-memory

File: docs/nd4j/templates/quickstart.md [hunk 25]
```diff
@@ -0,0 +1,565 @@
...
+ [    0.6117,    0.5385,    0.1251,    0.6886]]
+*/
+
+x2
+// [    0.1604,    0.0322,    0.8910,    0.4604,    0.7724,   -1.0000,    0.1617,    0.7586,    0.6117,    0.5385,    0.1251,    0.6886]
+```
+
+## Functions and Methods Overview
+
+### Array Creation
```
[the remark this answers is not in the document]
Comment (RobAltena) [reply docs/nd4j/templates/quickstart.md]: Yes, we can (and should) go deeper here. What I have so far is loop over the links in the Numpy quick start. I have removed functions that Nd4j did not have (Or rather: I could not find quickly).

**Pull request merged**